src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -356,6 +358,12 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
         .map(Optional::of);
   }
 
+  private final Function<Mono<BookingTO>, Mono<BookingTO>> createShipmentEventFromBookingTO =
```

Comment:
You can use an `UnaryOperator` here instead of `Function` as the input and output are the same.

Reply:
changed accordingly